src/main/java/org/lisoft/lsml/view_fx/controls/EquippedItemCell.java

```diff
@@ -318,22 +331,60 @@ private boolean onRemoveItem() {
         return safeCommand(this, stack, new CmdRemoveItem(msgd, loadout, component, getItem()), msgd);
     }
 
-    private void setupEngineRatingDropDown(boolean aPgiMode) {
+    private <T> boolean addIfEngineExists(final ObservableList<T> aList, final T aValue, final Integer aRating, final EngineType aType, final Faction aFaction) {
+        try {
+            if (aRating != null && aType != null && aFaction != null) {
+                ItemDB.getEngine(aRating, aType, aFaction);
+            }
+            aList.add(aValue);
```

Comment:
The behaviour here is unexpected, if any of the arguments are null, the rating is added to the list, I would expect it not to as no engine with null faction for example exists. However, the other suggestions should make all the null checks here go away which would resolve this.

Reply:
Done.